fs/copy.go

```diff
@@ -199,5 +201,22 @@ func openAndCopyFile(target, source string) error {
 	}
 	defer tgt.Close()
 
-	return copyFileContent(tgt, src)
+	var tgtWriter io.Writer = tgt
+	if runtime.GOOS != "linux" {
+		tgtWriter = onlyWriter{tgt}
+	}
+
+	buf := bufferPool.Get().(*[]byte)
```

Comment:
The buffer didn't even end up getting used before, should we even bother with it rather than checking OS. It could be possible for Go to implement copy optimizations on other platforms as well.

Reply:
Updated. Let's use `io.Copy` directly.